src/util/ConfigUtils.ts

```diff
@@ -168,3 +184,131 @@ export function modifyModulesNotMatchingTargetVersion(
+ *
+ * Before:
+ * "modules": {
+ *		"jquery.sap.script": {
+ *			"jQuery.sap.extend@1.58.0": {
+ *				"newModulePath": "sap/ui/thirdparty/jquery",
+ *				"newVariableName": "jQuery",
+ *				"functionName": "extend",
+ *				"replacer": "jQueryExtend",
+ *				"version": "^1.58.0"
+ *			},
+ *			"jQuery.sap.extend@1.60.0": {
+ *				"newModulePath": "sap/base/util/merge",
+ *				"newVariableName": "merge",
+ *				"replacer": "mergeOrObjectAssign",
+ *				"version": "1.60.0"
+ *			}
+ *		}
+ *	}
+ *
+ * E.g. when using version "latest"
+ * After:
+ * "modules": {
+ *		"jquery.sap.script": {
+ *			"jQuery.sap.extend": {
+ *				"newModulePath": "sap/base/util/merge",
+ *				"newVariableName": "merge",
+ *				"replacer": "mergeOrObjectAssign",
+ *				"version": "1.60.0"
+ *			},
+ *			"jQuery.sap.extend@1.58.0": {
+ *				"newModulePath": "sap/ui/thirdparty/jquery",
+ *				"newVariableName": "jQuery",
+ *				"functionName": "extend",
+ *				"replacer": "jQueryExtend",
+ *				"version": "^1.58.0"
+ *			},
+ *			"jQuery.sap.extend@1.60.0": {
+ *				"newModulePath": "sap/base/util/merge",
+ *				"newVariableName": "merge",
+ *				"replacer": "mergeOrObjectAssign",
+ *				"version": "1.60.0"
+ *			}
+ *		}
+ *	}
+ *
+ * @param {object} oConfig config to modify
+ * @param {string} targetVersion UI5 target version e.g. 1.58.0
+ * @returns {object} a copy of the config with closest versions to targetVersion of modules for
+ * keys which have an '@' in the name.
+ * E.g. keys ["jQuery.sap.extend@1.60.0", "jQuery.sap.extend@1.58.0"]
+ * will be ["jQuery.sap.extend@1.60.0", "jQuery.sap.extend@1.58.0", "jQuery.sap.extend"] where "jQuery.sap.extend" is a copy of the closest matching version,
+ * e.g. "1.60.0" for targetVersion "latest"
+ */
+export function mergeModulesWithMultipleVersions(
+	oConfig: object,
+	targetVersion: string
+) {
+	// validate input parameters
+	if (!oConfig) {
+		throw new Error("No config supplied for modification");
+	}
+
+	const oModifiedConfig = JSON.parse(JSON.stringify(oConfig));
+	if (Object.keys(oConfig).length === 0 || !targetVersion) {
+		return oModifiedConfig;
+	}
+
+	// create a copy of modules
+	const oModules = JSON.parse(JSON.stringify(oModifiedConfig["modules"]));
+	Object.keys(oModules).forEach(sModuleGroup => {
+		// get all keys with multiple versions
+		const multiVersionModuleKeys = Object.keys(
+			oModules[sModuleGroup]
+		).filter(sModule => sModule.includes("@"));
+		if (multiVersionModuleKeys.length > 0) {
+			// group multiple version modules by module
+			const groupByModule = {};
+			multiVersionModuleKeys.forEach(sKey => {
+				const sModule = sKey.split("@")[0];
+				if (!groupByModule[sModule]) {
+					groupByModule[sModule] = [];
+				}
+				groupByModule[sModule].push(sKey);
+			});
+
+			//get highest version
+			const closestModules = [];
+			Object.keys(groupByModule).forEach(sModule => {
+				let localClosest;
+
+				groupByModule[sModule].filter(sKey => {
```

Comment:
Filter callback does not return a boolean!

Reply:
done

Reply:
done